content/courses/program-security/account-data-matching.md

```diff
@@ -76,214 +85,135 @@ pub struct AdminConfig {
-those labs, we’ll show in this lab how a missing data validation check could
-allow the vault to be drained.
+<Callout type="info">
+  Remember to define your custom errors in your program's error enum:
 
-#### 1. Starter
-
-To get started, download the starter code from the `starter` branch of
-[this repository](https://github.com/Unboxed-Software/solana-account-data-matching).
-The starter code includes a program with two instructions and the boilerplate
-setup for the test file.
-
-The `initialize_vault` instruction initializes a new `Vault` account and a new
-`TokenAccount`. The `Vault` account will store the address of a token account,
-the authority of the vault, and a withdraw destination token account.
+```rust
+#[error_code]
+pub enum MyError {
+    #[msg("Only the current admin can perform this action")]
+    InvalidAdmin,
+}
+```
 
-The authority of the new token account will be set as the `vault`, a PDA of the
-program. This allows the `vault` account to sign for the transfer of tokens from
-the token account.
+</Callout>
 
-The `insecure_withdraw` instruction transfers all the tokens in the `vault`
-account’s token account to a `withdraw_destination` token account.
+By implementing these checks, you ensure that only the rightful admin can update
+the admin_config account, significantly improving your program's security.
 
-Notice that this instruction \***\*does\*\*** have a signer check for
-`authority` and an owner check for `vault`. However, nowhere in the account
-validation or instruction logic is there code that checks that the `authority`
-account passed into the instruction matches the `authority` account on the
-`vault`.
+## Lab: Securing a Vault Program
 
-```rust
-use anchor_lang::prelude::*;
-use anchor_spl::token::{self, Mint, Token, TokenAccount};
+In this lab, we'll create a simple "vault" program to demonstrate the importance
+of account data matching. We'll implement both an insecure and a secure version
+of a withdraw instruction to highlight the difference.
 
-declare_id!("Fg6PaFpoGXkYsidMpWTK6W2BeZ7FEfcYkg476zPFsLnS");
+### 1. Setup
 
-#[program]
-pub mod account_data_matching {
-    use super::*;
+Clone the starter code from the `starter` branch of
+[this repository](https://github.com/Unboxed-Software/solana-account-data-matching).
 
-    pub fn initialize_vault(ctx: Context<InitializeVault>) -> Result<()> {
-        ctx.accounts.vault.token_account = ctx.accounts.token_account.key();
-        ctx.accounts.vault.authority = ctx.accounts.authority.key();
-        ctx.accounts.vault.withdraw_destination = ctx.accounts.withdraw_destination.key();
-        Ok(())
-    }
+The starter code includes a program with two instructions and a boilerplate test
+file:
 
-    pub fn insecure_withdraw(ctx: Context<InsecureWithdraw>) -> Result<()> {
-        let amount = ctx.accounts.token_account.amount;
+1. `initialize_vault`: Creates a new `Vault` account and associated
+   `TokenAccount`.
```

Comment:
```suggestion
1. `initialize_vault`: Creates a new `Vault` account and related `TokenAccount`.
```

Want to avoid "associated `TokenAccount`" because we already have "associated token accounts" and my understanding is that not what is meant here. Please let me know if that's not correct.

Reply:
You are right my apologies!